pkg/kubelet/kubelet_pods.go

```diff
@@ -2328,3 +2361,68 @@ func (kl *Kubelet) cleanupOrphanedPodCgroups(pcm cm.PodContainerManager, cgroupP
 		go pcm.Destroy(val)
 	}
 }
+
+func (kl *Kubelet) runtimeClassSupportsRecursiveReadOnlyMounts(pod *v1.Pod) bool {
+	var runtimeClassName string
+	if pod.Spec.RuntimeClassName != nil {
+		runtimeClassName = *pod.Spec.RuntimeClassName
+	}
+	rtClasses := kl.runtimeState.runtimeClasses()
+	return runtimeClassSupportsRecursiveReadOnlyMounts(runtimeClassName, rtClasses)
+}
+
+// runtimeClassSupportsRecursiveReadOnlyMounts checks whether the runtime class supports recursive read-only mounts.
+// The kubelet feature gate is not checked here.
+func runtimeClassSupportsRecursiveReadOnlyMounts(runtimeClassName string, runtimeClasses []v1.RuntimeClass) bool {
+	for _, f := range runtimeClasses {
+		if f.Name == runtimeClassName {
+			if f.Features == nil {
+				return false
+			}
+			if f.Features.RecursiveReadOnlyMounts == nil {
+				return false
+			}
+			if !*f.Features.RecursiveReadOnlyMounts {
+				return false
+			}
+			return true
```

Comment:
could this be
```
return f.Features != nil && f.Features.RecursiveReadOnlyMounts != nil && *f.Features.RecursiveReadOnlyMounts

```
I think I have the cases right but check my work:
Features == nil -> false
Features.RROM == nil -> false
*Features.RROM == false -> false

Reply:
fixed